### PR TITLE
[MOB-1316] Add support for MPID

### PR DESCRIPTION
**Pull request by vbabenkoru**

### CODE REVIEW

File: iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java [hunk 1]
```diff
@@ -465,6 +471,11 @@ public ModuleRegistrationResponse processRegistrationRequest(ModuleRegistrationR
                         .setIsChecked(true)
                         .setDescription("If enabled, mParticle will attempt to coerce string attributes into scalar types (integer, boolean, and float).")
         );
+        eventSettings.add(
+                new TextSetting(SETTING_USER_ID_FIELD, "User ID field")
```
Comment (vbabenkoru) [inline iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:475]: I'm not sure how to make a drop-down out of this, like i've seen in other partner integrations..

Comment (vbabenkoru) [reply iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:479]: Looks like this is something mParticle has to do manually

File: iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java [hunk 2]
```diff
@@ -465,6 +475,11 @@ public ModuleRegistrationResponse processRegistrationRequest(ModuleRegistrationR
                         .setIsChecked(true)
                         .setDescription("If enabled, mParticle will attempt to coerce string attributes into scalar types (integer, boolean, and float).")
         );
+        eventSettings.add(
```
Comment (samdozor) [inline iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:478]: we will add this setting on our side - please remove it from here.

File: iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java [hunk 3]
```diff
@@ -771,16 +785,20 @@ public AudienceMembershipChangeResponse processAudienceMembershipChangeRequest(A
...
 
-    private void addUserIdentitiesToRequest(UserRequest request, List<UserIdentity> identities) {
+    private void addUserIdentitiesToRequest(UserRequest request, EventProcessingRequest processingRequest) {
+        List<UserIdentity> identities = processingRequest.getUserIdentities();
         if (identities != null) {
             for (UserIdentity identity : identities) {
                 if (identity.getType().equals(UserIdentity.Type.EMAIL)) {
                     request.email = identity.getValue();
-                } else if (identity.getType().equals(UserIdentity.Type.CUSTOMER)) {
+                } else if (identity.getType().equals(UserIdentity.Type.CUSTOMER) && !shouldUseMPID(processingRequest)) {
```
Comment (samdozor) [inline iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:794]: I'd recommend converting this method to something that maps the value of the setting onto various identities, such as the other identities. right now this only supports customer id and mpid

**Comment by vbabenkoru** [comment]
@samdozor Can we leave `customerId` and `mpid` just now? Adding other identities would require additional changes both here and in Kits.

File: iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java (hunk 2 above)
Comment (vbabenkoru) [reply iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:478]: Are the name and the values ok?

File: iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java (hunk 3 above)
Comment (vbabenkoru) [reply iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java:794]: Are there best practices on how to handle this? Should we use something like `userIdField.equals(userIdentity.getType().toString())`?

**Comment by samdozor** [comment]
@vbabenkoru sure, we can leave as just those two. You'll need to update the kits per the other PR, though.

**Pull request merged**